File: self.py
#!/usr/bin/python

tape = """print '#!/usr/bin/python'
print
print 'tape = ' + '"'*3 + tape + '"'*3
print
print tape
"""

print '#!/usr/bin/python'
print
print 'tape = ' + '"'*3 + tape + '"'*3
print
print tape

File: self.rb
#!/usr/bin/ruby

tape = <<EOT
puts '#!/usr/bin/ruby'
puts
puts 'tape = <<EOT'
puts tape
puts 'EOT'
puts
puts tape
puts
EOT

puts '#!/usr/bin/ruby'
puts
puts 'tape = <<EOT'
puts tape
puts 'EOT'
puts
puts tape
puts

